csharp/src/Ice/Connection.cs

```
@@ -1658,4 +1609,158 @@ private enum State
             Closed
         };
     }
+
+    public abstract class IPConnection : Connection
+    {
+        public System.Net.IPEndPoint? LocalAddress
+        {
+            get
+            {
+                // Cache the address to be able to access it after the socket is dispose
```

Would be helpful to describe why we are using these addresses after the underlying socket is disposed.

---

I'm not convinced it's really useful to cache them and it could end being confusing: it would return a non-null value on a closed connection if it was cached previously... but if it wasn't, it would return null.

---

So just throw ObjectDisposedException like Socket does?

---

There are 2 things here:

- do we want to forward or not exceptions from the underlying properties? If not, we catch/ignore and return null (that's what we do right now)
- cache or not the returned value

We could get rid of the caching but continue to catch exceptions and return null (which is somewhat more convenient for the user code).

---

Fixed, get rid of the caching, and keep ignoring exceptions